### ISSUE TITLE
Don't warn about conventions when implementing traits

### ISSUE THREAD

**Issue by jkcclemens** [intellij-rust/intellij-rust#1715]
If `SomeTrait` is in a crate I don't control, I can't fix its bad conventions, so don't warn me about them when I implement it.

```rust
extern crate not_my_crate;

use not_my_crate::SomeTrait;

struct MyStruct;

impl SomeTrait for MyStruct {
  // This will warn as follows
  // methods called `as_*` usually take self by reference or self by mutable reference; consider choosing a less ambiguous name
  // This isn't my choice; don't warn me about it
  fn as_i64(self) -> i64 {
    unimplemented!();
  }
}
```

**Comment by farodin91** [comment]
I think it would be better to annotate only traits or impl without a traitref

**Comment by matklad** [comment]
The relevant code is `src/main/kotlin/org/rust/ide/inspections/RsNamingInspection.kt`

**Comment by vlad20012** [comment]
To be more precise, the actual code is located [here](https://github.com/intellij-rust/intellij-rust/blob/d4a54c17c1dfffe7c7d2cbb8424b0eb126306a20/src/main/kotlin/org/rust/ide/inspections/RsNamingInspection.kt#L205) (and the tests [here](https://github.com/intellij-rust/intellij-rust/blob/d4a54c17c1dfffe7c7d2cbb8424b0eb126306a20/src/test/kotlin/org/rust/ide/inspections/RsNamingInspectionTest.kt#L351))

The idea is that we should not display warnings for methods defined in `impl Trait for S {}` at all. So if `el.owner` is `RsFunctionOwner.Impl` we should check that `owner.impl.traitRef == null` and don't run inspection otherwise
